app/models/megaphone/publisher.rb

```diff
@@ -66,17 +66,18 @@ def check_episodes(episodes)
 
         # check if it is uploaded yet
         # if not go looking for the DTR media version
-        if !megaphone_episode.delivery_status.uploaded?
+        status = megaphone_episode.delivery_status
+        if !status.uploaded?
           megaphone_episode.upload_audio!
-        end
-
         # check if it is uploaded, but not delivered - see if megaphone has processed
-        status = megaphone_episode.delivery_status
-        if !status.delivered? && status.uploaded?
+        elsif !status.delivered?
           megaphone_episode.check_audio!
         end
 
-        if !ep.episode_delivery_status(:megaphone).delivered?
+        # make sure we get the latest status on the episode
+        ep.reload
+        episode_status = ep.episode_delivery_status(:megaphone)
+        if !episode_status.delivered? || !episode_status.uploaded?
```

Comment:
Calling `reset` on delivery status, but saw some weird stuff where it seemed like the status wasn't updated - may not have been a thing, but this is just a bit more defensive about reloading and checking status.